### PR TITLE
WIP: fix GS import of content in Python 3 (#124)

### PR DESCRIPTION
**Pull request by tkimnguyen**
for #124

### CODE REVIEW

**Comment by tkimnguyen** [comment]
@jenkins-plone-org please run jobs

File: CHANGES.rst [hunk 1]
```diff
@@ -9,6 +9,15 @@ Changelog
 
 .. towncrier release notes start
 
+2.9.6 (unreleased)
+------------------
+
+Bug fixes:
+
+- fix GS import of content in Python 3 (#124)
+  [tkimnguyen]
```
Comment (gforcada) [inline CHANGES.rst]: that should go to a news file

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
While at Barceloneta Sprint the topic of GS export/import came up with @santonelli and so I found this PR. I rebased it and fixed the changelog to use towncrier. 

It looks like the change do not cover every case, since tests are failing.

**Comment by mauritsvanrees** [comment]
For me, importing an easyform in Plone 5.2 with Python 3, using `BytesIO` instead of `StringIO` helped. But the `decode` calls might be good too.

The Jenkins links have expired, but I see one failure locally with this PR, which is fixed with a single line change in the tests:

```
$ git diff
diff --git a/plone/dexterity/tests/test_exportimport.py b/plone/dexterity/tests/test_exportimport.py
index 2715da1..0194037 100644
--- a/plone/dexterity/tests/test_exportimport.py
+++ b/plone/dexterity/tests/test_exportimport.py
@@ -38,7 +38,7 @@ class ExportImportTests(unittest.TestCase):
         item = DummyItem('test')
 
         import_context = DummyImportContext(None)
-        import_context._files['.data'] = 'title: Foo'
+        import_context._files['.data'] = b'title: Foo'
         importer = DexterityContentExporterImporter(item)
         importer.import_(import_context, None, root=True)
``` 

So let the dummy import context return bytes when its mock `readDataFile` is called. This seems to match what I get in practice.

But I see this already goes wrong in GenericSetup, where our code is an adaptation from. So we may want to get it right there first. See https://github.com/zopefoundation/Products.GenericSetup/issues/107. And possibly related: https://github.com/zopefoundation/Products.GenericSetup/issues/85

Browsing through the code a bit, I get the feeling that `readDataFile` can return either bytes or string (using the Python 3 terms) or None, depending on the context (could also be a tarball). And some of the places where it gets used need bytes and some string. So for each line where we call `readDataFile` we should first determine what we need as output, and then in the code call encode/decode when needed, without relying on a specific return type from `readDataFile`.

**Comment by mauritsvanrees** [comment]
The same problem has now been fixed in GenericSetup: https://github.com/zopefoundation/Products.GenericSetup/pull/108
So we should look at that PR, and adapt our code to look the same.
(I don't have time/priority for that now.)

**Comment by mauritsvanrees** [comment]
I close this PR in favour of my own PR #152. I have taken over the GenericSetup changes there.

**Pull request closed without merging**